Lib/typing.py

```diff
@@ -2252,10 +2252,10 @@ def __new__(cls, name, bases, ns, total=True):
         Subclasses and instances of TypedDict return actual dictionaries.
         """
         for base in bases:
-            if type(base) is not _TypedDictMeta:
+            if not (type(base) is _TypedDictMeta or base is Generic):
                 raise TypeError('cannot inherit from both a TypedDict type '
                                 'and a non-TypedDict base class')
-        tp_dict = type.__new__(_TypedDictMeta, name, (dict,), ns)
+        tp_dict = type.__new__(_TypedDictMeta, name, bases + (dict,), ns)
```

Comment:
If we will add all classes to bases it will not be backward compatible.

So I think only `Generic` class should be added to generated class bases. 

 ```suggestion
        tp_bases = tuple(b for b in bases if b is Generic)
        tp_dict = type.__new__(_TypedDictMeta, name, (*tp_bases, dict), ns)
```